internal/cmd/headless_repl/src/processes/invoke_go.rs

```diff
@@ -0,0 +1,130 @@
+use crate::invoke_go::OutputError::InvalidMethod;
+use crate::processes::lsp_invoke::add_headers;
+use crate::processes::process_completion::process_completions_response;
+use crate::LSPSuggestionHelper::LSPSuggestionHelper;
+use crate::{CommandHint, MyHelper};
+use lsp_types::{
+    DidChangeTextDocumentParams, TextDocumentContentChangeEvent, Url,
+    VersionedTextDocumentIdentifier,
+};
+use regex::Regex;
+use rustyline::Helper;
+use serde_json::Value;
+use std::collections::HashSet;
+use std::env::consts::OS;
+use std::fmt::format;
+use std::hash::Hash;
+use std::io::{stdin, Read};
+use std::process::{Child, ChildStdout, Command, Stdio};
+use std::ptr::write;
+use std::str;
+use std::str::from_utf8;
+use std::string::String;
+use std::sync::atomic::AtomicUsize;
+use std::sync::mpsc::Sender;
+use std::sync::{Arc, Mutex, RwLock};
+use tower_lsp::jsonrpc;
+use tower_lsp::jsonrpc::RequestBuilder;
+
+pub fn start_go() -> Child {
+    let mut child = Command::new("./main")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .expect("failure to execute");
+    child
+}
+
+#[derive(Debug)]
+pub enum OutputError {
+    InvalidMethod,
+}
+
+impl From<serde_json::Error> for OutputError {
+    fn from(_: serde_json::Error) -> Self {
+        InvalidMethod
+    }
+}
+
+pub fn form_output(request_type: &str, text: &str) -> Result<String, OutputError> {
+    match request_type {
+        "Service.DidOutput" => {
+            let mut cleaned = text.to_string().replace("\"", "\\\"");
```

Comment:
```suggestion
            let mut cleaned = text.replace("\"", "\\\"");
```
`replace` exists on `&str`, no need to heap allocated a `String` https://doc.rust-lang.org/std/primitive.str.html#method.replace